traffic_ops/traffic_ops_golang/profileparameter/unassignedparams.go

```diff
@@ -32,24 +32,24 @@ import (
 
 // my $rs_data = $self->db->resultset("Parameter")->search( 'me.id' => { 'not in' => \@assigned_params } );
 
-func GetUnassigned(db *sql.DB) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		_, intParams, userErr, sysErr, errCode := api.AllParams(r, []string{"id"}, []string{"id"})
-		if userErr != nil || sysErr != nil {
-			api.HandleErr(w, r, errCode, userErr, sysErr)
-			return
-		}
-		api.RespWriter(w, r)(getUnassignedParametersByProfileID(intParams["id"], db))
+func GetUnassigned(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"id"}, []string{"id"})
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, errCode, userErr, sysErr)
+		return
 	}
+	defer inf.Close()
+	*inf.CommitTx = true
+	api.RespWriter(w, r)(getUnassignedParametersByProfileID(inf.Tx.Tx, inf.IntParams["id"]))
```

Comment:
Same Content-Type issue here, I think api.RespWriter or api.WriteResp should add the Content-Type json header.

Reply:
They do https://github.com/apache/trafficcontrol/blob/master/traffic_ops/traffic_ops_golang/api/api.go#L67